bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigOptionProvider.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.config.core;
+
+import java.util.Collection;
+import java.util.Locale;
+
+/**
+ * The {@link ConfigOptionProvider} can be implemented and registered as an <i>OSGi</i>
+ * service to provide {@link ConfigDescription}s options.
+ *
+ * @author Chris Jackson - Initial contribution
+ */
+public interface ConfigOptionProvider {
+
+    /**
+     * Provides a collection of {@link ParameterOptions}s.
+     *
+     * @param param
```

Comment:
So, this comes back to my question on the forum... At the moment, the way I've implemented this is that the ConfigOptionProvider is an interface that is implemented by a thingHandler (etc). So, it doesn't need the URI - the ThingConfigDescriptionProvider handles the URI, checks if the thingHandler implements the interface, and calls the method if so...
My previous implementation did have the URI here, but I don't think it's necessary with the ThingConfigDescriptionProvider?